### PR TITLE
[BPK-3168] Support phone-input masking

### PR DESCRIPTION
**Pull request by georgegillams**
Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] Docs - Follow up PR required to showcase this option

### CODE REVIEW

File: packages/bpk-component-phone-input/stories.js [hunk 1]
```diff
@@ -110,16 +113,17 @@ class StoryContainer extends Component<
...
           value={value}
           dialingCode={dialingCode}
           dialingCodes={[
-            { code: '1', description: '+1 (US)' },
-            { code: '2', description: '+1 (CA)' },
-            { code: '44', description: '+44 (UK)' },
-            { code: '55', description: '+55' },
-            { code: '998', description: '+998' },
+            { code: '1_us', description: '+1 (US)', numberPrefix: '+1' },
+            { code: '2_ca', description: '+2 (CA)', numberPrefix: '+2' },
```
Comment (ojcurtis) [inline packages/bpk-component-phone-input/stories.js]: Can you put these back to `1` as Canada has the same dialing code as US

File: packages/bpk-component-phone-input/src/BpkPhoneInput.js [hunk 2]
```diff
@@ -97,6 +99,49 @@ const BpkPhoneInput = (props: Props) => {
     disabled: !!disabled,
   };
 
+  const dialingCodeDefinition = dialingCodes.find(
+    dC => dC.code === dialingCode,
```
Comment (ojcurtis) [inline packages/bpk-component-phone-input/src/BpkPhoneInput.js]: Could you rename this to be more descriptive?

File: packages/bpk-component-calendar/src/__snapshots__/BpkCalendarGridTransition-test.js.snap [hunk 3]
```diff
@@ -17,13 +17,13 @@ exports[`BpkCalendar should render correctly 1`] = `
     }
   >
     <div>
-      {"minDate":"2009-02-01T00:00:00.000Z","maxDate":"2011-02-01T00:00:00.000Z","month":"2010-01-01T00:00:00.000Z","preventKeyboardFocus":true,"isKeyboardFocusable":false,"focusedDate":"2010-01-01T00:00:00.000Z","aria-hidden":true,"className":"bpk-calendar-grid-transition__grid"}
+      {"minDate":"2009-02-01T00:00:00.000Z","maxDate":"2011-02-01T00:00:00.000Z","month":"2010-01-01T00:00:00.000Z","preventKeyboardFocus":true,"isKeyboardFocusable":false,"focusedDate":"2010-01-01T01:00:00.000Z","aria-hidden":true,"className":"bpk-calendar-grid-transition__grid"}
```
Comment (ojcurtis) [inline packages/bpk-component-calendar/src/__snapshots__/BpkCalendarGridTransition-test.js.snap]: Is this change intentional?

File: packages/bpk-component-phone-input/stories.js [hunk 4]
```diff
@@ -26,11 +26,11 @@ import BpkImage from 'bpk-component-image';
 import BpkPhoneInput from './index';
 
 const DIALING_CODE_TO_ID_MAP = {
-  '1': 'us',
-  '2': 'ca',
-  '44': 'uk',
-  '55': 'br',
-  '998': 'uz',
+  '1_us': 'us',
+  '2_ca': 'ca',
```
Comment (ojcurtis) [inline packages/bpk-component-phone-input/stories.js]: Can you put these back to 1 as Canada has the same dialing code as US

File: packages/bpk-component-slider/src/__snapshots__/BpkSlider-test.js.snap [hunk 5]
```diff
@@ -12,33 +12,10 @@ exports[`BpkSlider should render correctly 1`] = `
   }
 >
   <div
-    className="bpk-slider__bar bpk-slider__bar-0"
```
Comment (ojcurtis) [inline packages/bpk-component-slider/src/__snapshots__/BpkSlider-test.js.snap]: Is this change intentional?

File: packages/bpk-component-slider/src/__snapshots__/BpkSlider-test.js.snap [hunk 6]
```diff
@@ -12,33 +12,10 @@ exports[`BpkSlider should render correctly 1`] = `
...
-      }
-    }
-  />
-  <div
-    aria-orientation="horizontal"
     aria-valuemax={100}
     aria-valuemin={0}
     aria-valuenow={25}
-    className="bpk-slider__handle bpk-slider__handle-0 "
+    className="handle handle-0 "
```
Comment (ojcurtis) [inline packages/bpk-component-slider/src/__snapshots__/BpkSlider-test.js.snap]: Is this right?

File: packages/bpk-component-slider/src/__snapshots__/BpkSlider-test.js.snap [hunk 7]
```diff
@@ -68,33 +45,10 @@ exports[`BpkSlider should render correctly with a "className" attribute 1`] = `
   }
 >
   <div
-    className="bpk-slider__bar bpk-slider__bar-0"
```
Comment (ojcurtis) [inline packages/bpk-component-slider/src/__snapshots__/BpkSlider-test.js.snap]: Same here

File: packages/bpk-component-calendar/src/__snapshots__/BpkCalendarGridTransition-test.js.snap (hunk 3 above)
Comment (georgegillams) [reply packages/bpk-component-calendar/src/__snapshots__/BpkCalendarGridTransition-test.js.snap]: No
